src/main/java/org/embulk/input/sftp/SftpFileInput.java

```diff
@@ -266,7 +266,7 @@ public boolean isRetryableException(Exception exception)
                         {
                             if (exception.getCause() != null && exception.getCause().getCause() != null) {
                                 Throwable cause = exception.getCause().getCause();
-                                if (cause.getMessage().contains("Auth fail")) {
+                                if (!cause.getMessage().isEmpty() && cause.getMessage().contains("Auth fail")) {
```

Comment:
Sorry for confusion 🙇 
What I mean is some exceptions won't have message, for example: `java.util.concurrent.TimeoutException`. So this line still could raise NPE.